src/memory/AmdFlash.cc

```diff
@@ -200,18 +179,13 @@ void AmdFlash::setState(State newState)
 {
 	if (state == newState) return;
 	state = newState;
-	motherBoard.getCPU().invalidateAllSlotsRWCache(0x0000, 0x10000);
+	config.getMotherBoard().getCPU().invalidateAllSlotsRWCache(0x0000, 0x10000);
 }
 
 uint8_t AmdFlash::peek(size_t address) const
 {
 	if (state == State::IDLE) {
-		auto [sector, sectorSize, offset] = getSectorInfo(address);
-		if (const uint8_t* addr = readAddress[sector]) {
-			return addr[offset];
-		} else {
-			return 0xFF;
-		}
+		return ram[address % size()];
```

Comment:
Here and in a few other places you added `% size()`. Why is that? Isn't it fine/better to assert that `ram` won't be accessed out-of-bounds?

Detail: (orthogonal) the return-type of `size()` is a `size_t`. Maybe it should be a `power_of_two`? So retaining this knowledge for longer, increasing the likelihood that the compiler optimizes such `%` operations to an `&`. 
A smart enough compiler may still see through this, so maybe this doesn't matter. On the other hand, also for human readers of the code, retaining the `power_of_two` knowledge might be useful.

Reply:
An address greater than the capacity is legit input IMO. Mirroring will be the hardware implementation for unused address lines, so I think it is correct to allow it. It’s better than adding `& (flash.size() - 1)` in all places that interact with AmdFlash.

For example take an ASCII16-X device with 8 MB Flash ROM, where the mapper can address up to 64 MB. The software can poke the mapper to a bank >= `0x200`, thus providing an address greater than the capacity. Currently it does not mirror, while it should, and I think it’s best to do that in AmdFlash itself.

It’s also defensive programming to make sure we don’t access out of bounds; a mistake would mean a security bug (asserts are compiled out of release builds). AmdFlash has nontrivial code that has been written with in-range addresses in mind. It’s easy to miss handling unexpected input, and hard to make guarantees. A class of bugs can be eliminated by making sure in the public interface that the input is never unexpected.

I’ll pull this out into a separate commit, including `power_of_two<size_t> size()`.